### DISCUSSION TITLE
Issue with contact force in pybullet during sliding

### DISCUSSION THREAD

**Discussion by deepakraina99** [bulletphysics/bullet3#3711] [General]
I have implemented an impedance control in pybullet. That means, whenever some external force is applied on a robot, it behaves as a spring and damper system. Due to this force, it gets displaced from its position and returns to its target position when an external force is removed. The equation that is used is given below:

![Screenshot from 2020-08-02 11-02-48](https://user-images.githubusercontent.com/34967469/89116281-bfc5d480-d4af-11ea-9cf7-189cc7cb9879.png)

The video regarding the same is given below:
![imp-ctrl-ur5-3](https://user-images.githubusercontent.com/34967469/89116086-a885e780-d4ad-11ea-9f23-9dd4a0ef8049.gif)

Now I am trying to implement the same control strategy in a scenario where robot is polishing a surface with some uncertainties (unevenness) on the surface. I moved the robot close to the table and try to move while it is in contact with table. I am getting the normal force during contact using the p.getContactPoints. But the robot suddenly takes a and leaves contact. As and when it again come in contact, it jumps again. The demo is shown below:
![ur5-imp](https://user-images.githubusercontent.com/34967469/89116826-ff8fba80-d4b5-11ea-99a6-b6bc78774fe6.gif)
I want robot to stay in continuous contact of this surface even when the table is moving upside down (say +/- 5cm). Please guide me what I am doing wrong here.
Thanks

**Comment by Steven89Liu** [comment]
Could you share your demo with me? thanks.

**Comment by marvinalles** [comment]
Hi @deepakraina99,

I would like to ask if you could share your impedance control implementation? I am trying to use cartesian impedance control for insertion, but somehow my implementation is not working properly. I would be really grateful!

Thank you!
